src/test/java/com/conveyal/gtfs/validator/FlexValidatorTest.java

```diff
@@ -100,192 +141,285 @@ private static Stream<LocationArguments> createLocationChecks() {
+            )
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("createDepartureTimeTests")
+    void validateDepartureTime(BaseArguments baseArguments) {
+        List<NewGTFSError> errors = new ArrayList<>();
+        FlexValidator.validateDepartureTime((StopTime) baseArguments.testObject, errors);
+        checkValidationErrorsMatchExpectedErrors(errors, baseArguments.expectedErrors);
+    }
+
+    private static Stream<BaseArguments> createDepartureTimeTests() {
+        return Stream.of(
+            new BaseArguments(
+                createStopTimeForDepartureTimeTest(1130, INT_MISSING, INT_MISSING, INT_MISSING),null
+            ),
+            new BaseArguments(
+                createStopTimeForDepartureTimeTest(1130, INT_MISSING, 1200, INT_MISSING),
+                Lists.newArrayList(NewGTFSErrorType.FLEX_FORBIDDEN_DEPARTURE_TIME)
             ),
             new BaseArguments(
-                createStopTime(INT_MISSING, INT_MISSING, 1100, INT_MISSING),null
+                createStopTimeForDepartureTimeTest(1130, INT_MISSING, INT_MISSING, 1200),
+                Lists.newArrayList(NewGTFSErrorType.FLEX_FORBIDDEN_DEPARTURE_TIME)
+            )
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("createStopIdTests")
+    void validateStopId(BaseArguments baseArguments) {
+        List<NewGTFSError> errors = new ArrayList<>();
+        FlexValidator.validateStopId((StopTime) baseArguments.testObject, errors);
+        checkValidationErrorsMatchExpectedErrors(errors, baseArguments.expectedErrors);
+    }
+
+    private static Stream<BaseArguments> createStopIdTests() {
+        return Stream.of(
+            new BaseArguments(
+                createStopTimeForStopIdTest("stop-id-1", null, null),null
             ),
             new BaseArguments(
-                createStopTime(INT_MISSING, INT_MISSING, INT_MISSING, 1200),null
+                createStopTimeForStopIdTest(null, null, null),
+                Lists.newArrayList(NewGTFSErrorType.FLEX_REQUIRED_STOP_ID)
             ),
             new BaseArguments(
-                createStopTime(INT_MISSING, INT_MISSING, 1100, 1200),null
+                createStopTimeForStopIdTest("stop-id-1", "location-group-id-1", null),
+                Lists.newArrayList(NewGTFSErrorType.FLEX_FORBIDDEN_STOP_ID)
             ),
             new BaseArguments(
-                createStopTime(1130, INT_MISSING, 1100, INT_MISSING),
-                Lists.newArrayList(NewGTFSErrorType.FLEX_FORBIDDEN_ARRIVAL_TIME, NewGTFSErrorType.FLEX_FORBIDDEN_START_PICKUP_DROP_OFF_WINDOW)
+                createStopTimeForStopIdTest("stop-id-1", null, "location-id-1"),
+                Lists.newArrayList(NewGTFSErrorType.FLEX_FORBIDDEN_STOP_ID)
+            )
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("createLocationGroupIdTests")
+    void validateLocationGroupId(BaseArguments baseArguments) {
```

Comment:
You could spread `baseArguments` and put its attributes as this method's arguments instead.